### PR TITLE
bpo-30109: Fix reindent.py

### PR DESCRIPTION
**Pull request by Mariatta**
Skip the file if it has bad encoding.

https://bugs.python.org/issue30109

### CODE REVIEW

File: Lib/test/test_tools/test_reindent.py [hunk 1]
```diff
@@ -23,6 +23,13 @@ def test_help(self):
         self.assertEqual(out, b'')
         self.assertGreater(err, b'')
 
+    def test_reindent_file_with_bad_encoding(self):
+        bad_coding_path = os.path.join(
+            os.path.dirname(os.path.realpath(__file__)), '../bad_coding.py')
```
Comment (serhiy-storchaka) [inline Lib/test/test_tools/test_reindent.py]: `os.path.join(..., os.pardir, 'bad_coding.py')`

It is common also to use `support.findfile()`.

File: Lib/test/test_tools/test_reindent.py [hunk 2]
```diff
@@ -23,6 +23,13 @@ def test_help(self):
...
+        rc, out, err = assert_python_ok(self.script, '-r', bad_coding_path)
+        self.assertEqual(out, b'')
+        self.assertGreater(err, b'')
```
Comment (serhiy-storchaka) [inline Lib/test/test_tools/test_reindent.py]: I think `assertNotEqual()` is more appropriate here. It produces more detailed report for strings and perhaps will produce more detailed report for bytes in future.

File: Lib/test/test_tools/test_reindent.py (hunk 1 above)
Comment (Mariatta) [reply Lib/test/test_tools/test_reindent.py]: Thanks :) I changed to `findfile()`.

File: Lib/test/test_tools/test_reindent.py (hunk 2 above)
Comment (Mariatta) [reply Lib/test/test_tools/test_reindent.py]: Changed to `assertNotEqual` :)

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by Mariatta** [comment]
Thanks @serhiy-storchaka :)

**Comment by Mariatta** [comment]
Backport to 2.7 has been done in https://github.com/python/cpython/pull/5637.

**Pull request merged**